### PR TITLE
fix(shorebird_cli): auth token refresh

### PR DESCRIPTION
**Pull request by felangel**
## Description

- fix(shorebird_cli): auth token refresh

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [X] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/auth/auth.dart [hunk 1]
```diff
@@ -28,22 +28,48 @@ final _clientId = ClientId(
...
   void Function(String) userPrompt,
 );
 
+typedef RefreshCredentials = Future<oauth2.AccessCredentials> Function(
+  oauth2.ClientId clientId,
+  oauth2.AccessCredentials credentials,
+  http.Client client,
+);
+
+typedef OnRefreshCallback = void Function(oauth2.AccessCredentials credentials);
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/auth/auth.dart]: naming nit: `OnRefreshCredentials`

File: packages/shorebird_cli/test/src/auth/auth_test.dart [hunk 2]
```diff
@@ -38,11 +42,79 @@ void main() {
...
+          onRefreshCallback: refreshCallbacks.add,
+        );
+
+        await client.get(Uri.parse('https://example.com'));
+
+        expect(refreshCallbacks, isEmpty);
+        final captured = verify(() => httpClient.send(captureAny())).captured;
+        expect(captured, hasLength(1));
+        final request = captured.first as http.BaseRequest;
+        expect(request.headers['Authorization'], equals('Bearer $idToken'));
```
Comment (bryanoltman) [inline packages/shorebird_cli/test/src/auth/auth_test.dart:114]: Should we also verify that the refresh callback was never invoked?

**Review by bryanoltman — APPROVED** [review]

File: packages/shorebird_cli/test/src/auth/auth_test.dart (hunk 2 above)
Comment (felangel) [reply packages/shorebird_cli/test/src/auth/auth_test.dart:114]: I do that on line 110

File: packages/shorebird_cli/lib/src/auth/auth.dart (hunk 1 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/auth/auth.dart]: Good call, renamed

File: packages/shorebird_cli/test/src/auth/auth_test.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/test/src/auth/auth_test.dart:114]: Missed it! Ignore me :)

Comment (felangel) [reply packages/shorebird_cli/test/src/auth/auth_test.dart:114]: No worries glad you asked!

**Pull request merged**